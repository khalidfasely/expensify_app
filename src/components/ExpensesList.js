import React from 'react';
import { connect } from 'react-redux';
import ExpenseItem from './ExpenseListItem';
import selectorExpense from '../selectors/expenses';

export const ExpensesList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No expenses</p>
            ) : (
                props.expenses.map((expense) => {
                return <ExpenseItem key={expense.id} {...expense} />
            })
            )
        }
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectorExpense(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpensesList);