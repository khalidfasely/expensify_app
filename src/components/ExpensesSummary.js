import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectorExpense from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expensesCount, expensesCountTotal }) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesCountTotal / 100).format('$0,0.00');
    
    return (
    <div>
        <h1>Viewing {expensesCount} { expenseWord }, totalling {formattedExpensesTotal}</h1>
    </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectorExpense(state.expenses, state.filters);
    return {
        expensesCount: visibleExpenses.length,
        expensesCountTotal: expensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);