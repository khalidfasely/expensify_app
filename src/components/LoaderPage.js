import React from 'react';

const LoaderPage = () => (
    <div className="loader">
        <img className="loader__image" src="/images/loader.gif"/>
    </div>
);

export default LoaderPage;