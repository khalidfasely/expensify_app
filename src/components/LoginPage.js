import React from 'react';
import { connect } from 'react-redux';
import { startLogin, startLoginWithGitHub } from '../actions/auth';

export const Login = ({ startLogin, startLoginWithGitHub }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>it's time to get your expenses under control.</p>
            <button className="button button--login google_login" onClick={startLogin}>
                Login with Google
            </button>
            <button className="button button--login github_login" onClick={startLoginWithGitHub}>
                Login with GitHub
            </button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin()),
    startLoginWithGitHub: () => dispatch(startLoginWithGitHub())
});

export default connect (undefined, mapDispatchToProps)(Login);

//<img className="icon" src="./images/google.png" alt='google icon' />
//<img className="icon" src={githubIcon} alt='github icon' />