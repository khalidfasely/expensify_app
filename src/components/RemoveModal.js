import React from 'react';
import Modal from 'react-modal';

const RemoveModal = (props) => (
    <Modal
        isOpen={props.modalOpen}
        contentLabel="Selected Option"
        onRequestClose={props.handleClearRemoveModal}
        ariaHideApp={false}
        closeTimeoutMS={200}
        className="modal"
    >
        <h3 className="modal__title">Remove The Expense</h3>
        <button className="button--modal button--modal--y" onClick={props.onRemove}>Yes</button>
        <button className="button--modal button--modal--n" onClick={props.handleClearRemoveModal}>No</button>
    </Modal>
);

export default RemoveModal;