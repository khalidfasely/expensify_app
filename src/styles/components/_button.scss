.button {
    background: $blue;
    border: none;
    color: white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

.button--login {
    margin-bottom: $s-size;
    min-width: 20rem;
    max-width: 25rem;
}

.google_login {
    background: lighten(white, 10%);
    color: black;
    box-shadow: 2px 2px 4px $grey;
}

.github_login {
    background: lighten(#0d1117, 15%);
    box-shadow: 2px 2px 4px $grey;
}

.button--link {
    background: none;
}

.button--secondary {
    background-color: #888;
    margin-bottom: 1rem;
}

.button--modal {
    @extend .button;
    font-weight: 700;
    background: none;
    margin: 0 $s-size;
}

.button--modal--y {
    color: lighten(green, 10%);
}

.button--modal--n {
    color: darken(red, 5%);
}