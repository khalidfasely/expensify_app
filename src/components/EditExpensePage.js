import React from 'react';
import { connect } from 'react-redux';
import RemoveModal from './RemoveModal';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
  state = {
    modalOpen: false
  };

  componentDidMount() {
    this.setState({ modalOpen: true });
    setTimeout(() => {
      this.setState({ modalOpen: false });
    }, 50)
  }

  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };
  onClick = () => {
    this.setState(() => ({ modalOpen: true }));
  };
  onRemove = () => {
    this.handleClearRemoveModal();
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  };
  handleClearRemoveModal = () => {
    this.setState(() => ({ modalOpen: false }));
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
            history={this.props.history}
          />
          <button className=" button button--secondary" onClick={this.onClick}>Remove Expense</button>
        </div>
        <RemoveModal
          onClick={this.onClick}
          modalOpen={this.state.modalOpen}
          onRemove={this.onRemove}
          handleClearRemoveModal={this.handleClearRemoveModal}
        />
      </div>
    )
  }
};

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  };
};

const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
//export default connect(mapStateToProps)(EditExpensePage);