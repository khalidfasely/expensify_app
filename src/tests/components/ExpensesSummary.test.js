import React from 'react';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';

test('should render ExpensesSummary with 1 expense correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with 2 or more expenses correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});