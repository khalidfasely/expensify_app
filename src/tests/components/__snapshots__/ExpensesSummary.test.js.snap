// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 expense correctly 1`] = `
<div>
  <h1>
    Viewing 
     
    expenses
    , totalling 
    $0.00
  </h1>
</div>
`;

exports[`should render ExpensesSummary with 2 or more expenses correctly 1`] = `
<div>
  <h1>
    Viewing 
     
    expenses
    , totalling 
    $0.00
  </h1>
</div>
`;
